File: EPZEU/.NET/EPZEU.Signing/Signing/Models/BtrustUserInputRequest.cs
namespace EPZEU.Signing.Models
{
    public class BtrustUserInputRequest
    {
        public string Input { get; set; }

        public BtrustUserInputTypes? InputType { get; set; }

        public string Otp { get; set; }

        public string CreateRpToClientAuthorizationHeder()
        {
            if (string.IsNullOrEmpty(Input) || InputType == null || InputType == BtrustUserInputTypes.PROFILE)
                return null;

            string headerValue = string.Format("personalId:{0}"
                , InputType.Value == BtrustUserInputTypes.LNCH ? string.Format("PI:BG-{0}", Input): Input);

            return headerValue;
        }

        public string CreateClientTokenHeader(string clientToken)
        {
            string tokenHeaderValue = string.Format("clientToken:{0}", clientToken);

            return tokenHeaderValue;
        }
    }

    public enum BtrustUserInputTypes
    {
        EGN = 0,

        LNCH = 1,

        PROFILE = 2
    }
}

File: EPZEU_PR/property-register/pr-web/pr-web-public/src/main/webapp/app/EPZEU.Core/Models/BtrustUserInputRequest.ts
import { TypeSystem } from 'Cnsys.Core';
import { moduleContext } from '../ModuleContext';

export interface BtrustUserInputRequest {
    input: string;
    inputType?: BtrustUserInputTypes;
    otp?: string;
}

export enum BtrustUserInputTypes {
    EGN = 0,
    LNCH = 1,
    PROFILE = 2
}
TypeSystem.registerEnumInfo(BtrustUserInputTypes, 'BtrustUserInputTypes', moduleContext.moduleName);